src/core_plugins/tile_map/public/editors/wms_options.js

```diff
@@ -0,0 +1,11 @@
+import { uiModules } from 'ui/modules';
+import wmsOptionsTemplate from './wms_options.html';
+const module = uiModules.get('kibana');
+
+module.directive('wmsOptions', function () {
+  return {
+    restrict: 'E',
+    template: wmsOptionsTemplate,
+    replace: true
```

Comment:
Now looking back at it, not sure if I 100% understand the advantage. we want to merge these vis.param.wms-settings into the existing options of the parent-scope, just like all other settings. If not, don't we need custom way of handling merging parameters from the saved-object/uitState into this UI-component, something which happens transparently now?

Reply:
not really ... you pass your object down to directive, and directive can manipulate it .... right ?
the advantage is just easier to read code ... its would be clear what objects that directive needs and is manipulating, as right now it could change anything on the parents $scope.